content/terraform/v1.12.x/docs/language/manage-sensitive-data.mdx

```diff
@@ -0,0 +1,287 @@
+
+If you are developing with Terraform locally, Terraform stores your state in a plaintext file, which includes any secret values you defined in your configuration. Treat your state file as sensitive data by excluding it from Git workflows and following our recommendations to [secure your state file](#state-security-best-practices).
+
+You can also configure Terraform to store your state and plan files in a [remote location](/terraform/language/state/remote). When you configure Terraform to use remote state storage, it only holds state in memory when actively using it. You can encrypt your state at rest, but the encryption method depends on your specific backend. For example, HCP Terraform automatically encrypts state at rest and protects it with TLS in transit.
+
+## Requirements
+
+The following requirements are tied to specific blocks and arguments you can use to handle sensitive values in your Terraform configuration:
+
+- Use Terraform 0.15 or later to add the `sensitive` argument to `variable` and `output` blocks
+- Use Terraform 1.10 or later to add the `ephemeral` argument to variables and child module outputs, or add the `ephemeral` block
+- Use Terraform 1.11 or later to use a write-only argument on a managed resource
+
+Each provider defines any available `ephemeral` blocks and write-only arguments on their managed resources. Reference the [Terraform registry](https://registry.terraform.io) to learn more about what your provider supports.
+
+## Determine how to handle your sensitive data
+
+To decide how to handle sensitive values in your configuration, start by determining whether you want to hide values from the Terraform CLI and HCP Terraform UI, or prevent Terraform from storing those values entirely.
+
+You can add the `sensitive` argument to `variable` or `output` blocks to make Terraform redact those values from CLI output and in the HCP Terraform UI. Terraform also treats any expressions that reference a sensitive variable or output as inherently sensitive. Learn more [about sensitive variables and outputs](#hide-sensitive-variables-and-outputs).
+
+Ephemeral values are available at runtime, but Terraform omits them from state and plan files entirely. Terraform provides four ways to define ephemeral values in your configuration:
+
+- The `ephemeral` argument on variables and child module outputs
+- The `ephemeral` block
+- A write-only argument on a managed resource
+
+Learn more about [ephemeral values and their restrictions](#omit-values-from-state-and-plan-files).
+
+You can also add both the `ephemeral` and `sensitive` arguments to `variable` and child module `output` blocks to prevent Terraform from storing their values and redacting those values in the Terraform CLI and HCP Terraform UI. Learn more about [hiding values in the CLI without storing them](#hide-values-without-storing-them).
```

Comment:
```suggestion
You can also add both the `ephemeral` and `sensitive` arguments to `variable` and child module `output` blocks to prevent Terraform from storing their values. Doing so redacts those values in the Terraform CLI and HCP Terraform UI. Learn more about [hiding values in the CLI without storing them](#hide-values-without-storing-them).
```
Right? The goal is not to _prevent_ Terraform from redacting the values, you _want_ it to redact the values.